compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs

```diff
@@ -403,6 +403,7 @@ fn codegenned_and_inlined_items(tcx: TyCtxt<'_>) -> DefIdSet {
     let mut result = items.clone();
 
     for cgu in cgus {
+        #[allow(rustc::potential_query_instability)]
```

Comment:
Can you try turning the `items` field in `CodegenUnit` into an `FxIndexMap`. That should make iteration stable (if the collection is built in a deterministic order). An `UnordMap` would be even better, but I suspect that that would be more work.